<launch>
    <!-- İyileştirilmiş Exploration Node -->
    <node pkg="enkaz_rescue" type="exploration_node" name="apartment_exploration_node" output="screen">
        <!-- Temel topic ve frame parametreleri -->
        <param name="octomap_topic" value="/octomap_full" />
        <param name="goal_topic" value="/exploration_goal" />
        <param name="global_frame" value="world" />
        <param name="robot_base_frame" value="firefly/base_link" />
        
        <!-- Güncelleme frekansı -->
        <param name="update_frequency" value="0.5" />  <!-- 0.5 Hz - 2 saniyede bir -->
        
        <!-- APARTMAN SINIRLARI - ÖNEMLİ -->
        <param name="min_x" value="-8.5" />
        <param name="max_x" value="9.0" />
        <param name="min_y" value="1.0" />
        <param name="max_y" value="11.5" />
        <param name="min_z" value="0.5" />   <!-- Yerden biraz yukarı -->
        <param name="max_z" value="1.7" />   <!-- DENEME İÇİN 1.7m YÜKSEKLİK SINIRI -->
        
        <!-- Frontier kümeleme parametreleri -->
        <param name="min_cluster_size" value="25" />  <!-- Minimum küme boyutu (artırıldı) -->
        <param name="cluster_radius" value="0.8" />   <!-- Kümeleme yarıçapı (metre) -->
        
        <!-- Keşif değerlendirme parametreleri -->
        <param name="min_exploration_value" value="0.3" />  <!-- Minimum keşif değeri eşiği -->
        <param name="corner_penalty" value="0.4" />         <!-- Köşelere verilen ceza (0-1 arası) -->
        <param name="min_unexplored_ratio" value="0.4" />   <!-- En az %40 keşfedilmemiş alan olmalı -->
        
        <!-- Güvenli hedef belirleme parametreleri -->
        <param name="safe_distance_from_frontier" value="1.0" />  <!-- Frontier'dan güvenli mesafe -->
        <param name="exploration_radius" value="2.0" />           <!-- Keşfedilmişlik kontrol yarıçapı -->
    </node>

    <!-- İyileştirilmiş 3D Path Planner -->
    <node name="path_planner_3d" pkg="enkaz_rescue" type="path_planner_3d_node"
          output="screen" launch-prefix="bash -c 'sleep 3; $0 $@'">
        
        <!-- Temel parametreler -->
        <param name="global_frame" value="world" />
        <param name="robot_base_frame" value="firefly/base_link" />
        <param name="planning_timeout" value="5.0" />
        
        <!-- Robot boyutları - drone için gerçekçi değerler -->
        <param name="robot_radius" value="0.4" />   <!-- Drone yarıçapı (arttırıldı) -->
        <param name="safety_margin" value="0.1" />  <!-- Güvenlik marjı (arttırıldı) -->
        
        <!-- Path yumuşatma parametreleri -->
        <param name="interpolation_points" value="300" />
        <param name="goal_tolerance" value="0.15" />
        
        <!-- Planlayıcı tipi ve optimizasyon -->
        <param name="planner_type" value="RRTstar" />
        <param name="smoothing_time" value="1.5" />
        <param name="shortcut_attempts" value="100" />
        <param name="max_segment_length" value="0.3" />
        <param name="collision_check_resolution" value="0.05" />
        
        <!-- Planlama sınırları - Exploration ile uyumlu -->
        <param name="bounds/x_min" value="-8.5" />
        <param name="bounds/x_max" value="9.0" />
        <param name="bounds/y_min" value="1.0" />
        <param name="bounds/y_max" value="11.5" />
        <param name="bounds/z_min" value="0.3" />
        <param name="bounds/z_max" value="2.0" />  <!-- Path planner biraz daha yükseğe izin verebilir -->
    </node>

    <!-- Path Follower -->
    <node name="path_follower" pkg="enkaz_rescue" type="path_follower_node"
          output="screen" launch-prefix="bash -c 'sleep 6; $0 $@'">
        
        <!-- Path takip parametreleri -->
        <param name="waypoint_tolerance" value="0.3" />
        <param name="max_velocity" value="0.8" />       <!-- Daha yavaş, güvenli hız -->
        <param name="max_acceleration" value="0.4" />
        <param name="look_ahead_distance" value="1.2" />
        
        <!-- Yükseklik kontrolü -->
        <param name="min_altitude" value="0.5" />
        <param name="max_altitude" value="1.8" />  <!-- Follower da yükseklik sınırına uymalı -->
    </node>
    
    
    <!-- Debug modda çalıştırmak için -->
    <arg name="debug" default="false" />
    <group if="$(arg debug)">
        <env name="ROSCONSOLE_CONFIG_FILE" value="$(find enkaz_rescue)/config/rosconsole.conf"/>
    </group>
    
</launch>